test/integration/EndToEnd.t.sol

```diff
@@ -558,322 +505,188 @@ contract EndToEndFlows is EndToEndUtils {
         );
 
-        vm.startPrank(poolManager);
-        uint32 issueEpochId = hub.batchRequestManager.nowIssueEpoch(poolId, shareClassId, assetId);
-        (D18 sharePrice,) = hub.shareClassManager.pricePoolPerShare(poolId, shareClassId);
-        hub.batchRequestManager.issueShares{value: GAS}(
-            poolId, shareClassId, assetId, issueEpochId, sharePrice, HOOK_GAS, REFUND
+        vm.startPrank(FM);
+        uint32 issueEpochId = h.batchRequestManager.nowIssueEpoch(POOL_A, SC_1, s.usdcId);
+        (D18 sharePrice,) = h.shareClassManager.pricePoolPerShare(POOL_A, SC_1);
+        h.batchRequestManager.issueShares{value: GAS}(
+            POOL_A, SC_1, s.usdcId, issueEpochId, sharePrice, HOOK_GAS, REFUND
         );
     }
 
-    function _processAsyncDepositClaim(
-        CHub memory hub,
-        CSpoke memory spoke,
-        PoolId poolId,
-        ShareClassId shareClassId,
-        AssetId assetId,
-        address investor,
-        IAsyncVault vault,
-        uint128 amount
-    ) internal {
+    function _processAsyncDepositClaim(IAsyncVault vault, uint128 amount) internal {
         vm.startPrank(ANY);
         vm.deal(ANY, GAS);
-        hub.batchRequestManager.notifyDeposit{value: GAS}(
-            poolId,
-            shareClassId,
-            assetId,
-            investor.toBytes32(),
-            hub.batchRequestManager.maxDepositClaims(poolId, shareClassId, investor.toBytes32(), assetId),
+        h.batchRequestManager.notifyDeposit{value: GAS}(
+            POOL_A,
+            SC_1,
+            s.usdcId,
+            INVESTOR_A.toBytes32(),
+            h.batchRequestManager.maxDepositClaims(POOL_A, SC_1, INVESTOR_A.toBytes32(), s.usdcId),
             REFUND
         );
 
-        vm.startPrank(investor);
-        vault.mint(vault.maxMint(investor), investor);
+        vm.startPrank(INVESTOR_A);
+        vault.mint(vault.maxMint(INVESTOR_A), INVESTOR_A);
 
         // CHECKS
-        assertEq(
-            spoke.spoke.shareToken(poolId, shareClassId).balanceOf(investor), assetToShare(amount), "expected shares"
-        );
+        assertEq(s.spoke.shareToken(POOL_A, SC_1).balanceOf(INVESTOR_A), assetToShare(amount), "expected shares");
     }
 
     function _testAsyncDeposit(bool sameChain, bool nonZeroPrices) public {
         _configurePool(sameChain);
-        _asyncDepositFlow(h, s, POOL_A, SC_1, s.usdcId, FM, INVESTOR_A, USDC_AMOUNT_1, nonZeroPrices);
+        _configurePricesForFlow(nonZeroPrices);
+
+        vm.startPrank(FM);
+        h.hub.updateVault{value: GAS}(
+            POOL_A, SC_1, s.usdcId, s.asyncVaultFactory, VaultUpdateKind.DeployAndLink, EXTRA_GAS, REFUND
+        );
+        vm.stopPrank();
+        IAsyncVault vault = IAsyncVault(address(s.vaultRegistry.vault(POOL_A, SC_1, s.usdcId, s.asyncRequestManager)));
+
+        _executeAsyncDepositRequest(vault, USDC_AMOUNT_1);
+        _processAsyncDepositApproval(USDC_AMOUNT_1);
+        _processAsyncDepositClaim(vault, USDC_AMOUNT_1);
```

Comment:
These calls (and the homologous in `_testAsyncRedeem` / `_testSyncDeposit`) are only called once from here, wondering if we should unroll them here, to easily follow the steps along with the debugger.

For me it makes sense but want you opinion

Reply:
Ah yes you are right. Then no objection from my side.

Reply:
Ok, if no one else has strong opinions agains this, this is the diff: [08499eb](https://github.com/centrifuge/protocol/pull/704/commits/08499ebb77eb7dc6504a77516d3f70b1715b0ab4)